### PR TITLE
cnf-feature-deploy integration comments 

### PR DESCRIPTION
**Pull request by edcdavid**
This PR addresses comments from cnf-feature-deploy team (see: https://github.com/openshift-kni/cnf-features-deploy/pull/1282)
- renamed utils package with pkg
- created new packages to lighten ptp.go. Lots of support function were moved out of ptp.go in other packages (under pkg). Also removed gomega/ginkgo unnecessary dependencies in some functions.
- ported 3 cnf-feature-deploy ptp tests in ptp-operator ("Metrics reported by PTP pods", "Run pmc in a new pod on the slave node", "Negative - run pmc in a new unprivileged pod on the slave node")
- removed os.exit calls

The goal is to be able to run ptp conformance tests from within cnf-features-deploy code similarly to what is happening now but with the new ptp test framework.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,24 +2,33 @@ module github.com/openshift/ptp-operator
...
 	github.com/Masterminds/sprig v2.22.0+incompatible
 	github.com/go-logr/logr v1.2.3
 	github.com/golang/glog v1.0.0
 	github.com/onsi/ginkgo v1.16.5
-	github.com/onsi/gomega v1.20.2
-	github.com/openshift/api v0.0.0-20220831183848-09c070622e2c
-	github.com/openshift/client-go v0.0.0-20220831193253-4950ae70c8ea
-	github.com/openshift/library-go v0.0.0-20220922140741-7772048e4447
+	github.com/onsi/gomega v1.21.1
+	github.com/openshift-kni/cnf-features-deploy v0.0.0-20221007071024-22a40aaee43b
```
Comment (aneeshkp) [inline go.mod:19]: Why ptp operator has dependency of cnf-features-deploy?

File: go.mod [hunk 2]
```diff
@@ -2,24 +2,33 @@ module github.com/openshift/ptp-operator
 
 go 1.18
 
+replace k8s.io/client-go => k8s.io/client-go v0.25.2
```
Comment (josephdrichard) [inline go.mod:5]: Why are these specific versions required?

File: test/conformance/ptp/prometheus.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
+package ptp
```
Comment (josephdrichard) [inline test/conformance/ptp/prometheus.go:1]: What does this file do?

File: go.mod (hunk 1 above)
Comment (edcdavid) [reply go.mod]: fixed, pulled in images.go

File: go.mod (hunk 2 above)
Comment (edcdavid) [reply go.mod]: fixed, was due to a pevious conflict

File: test/conformance/ptp/prometheus.go (hunk 3 above)
Comment (edcdavid) [reply test/conformance/ptp/prometheus.go:1]: This supports one of the test imported by cnf-features-deploy. I get the metrics from ptp anf check if they are in prometheus locally

**Comment by josephdrichard** [comment]
Why are you moving things from utils to pkg?

**Comment by edcdavid** [comment]
> Why are you moving things from utils to pkg?
This is to address this comment:
https://github.com/openshift-kni/cnf-features-deploy/pull/1282#discussion_r990115038

**Comment by aneeshkp** [comment]
running locally error
Failed to compile conformance:

# github.com/openshift/ptp-operator/test/conformance/ptp
test/conformance/ptp/ptp.go:211:180: ptpInterface.IfPci.Description undefined (type libexports.PCIAddress has no field or method Description)

**Comment by edcdavid** [comment]
> running locally error Failed to compile conformance:
> 
> github.com/openshift/ptp-operator/test/conformance/ptp test/conformance/ptp/ptp.go:211:180: ptpInterface.IfPci.Description undefined (type libexports.PCIAddress has no field or method Description)
try the following:
go clean --modcache
go get github.com/test-network-function/l2discovery-exports
go mod tidy
go mod vendor

File: go.mod [hunk 4]
```diff
@@ -75,7 +75,7 @@ require (
 	github.com/prometheus/procfs v0.7.3 // indirect
 	github.com/spf13/pflag v1.0.5 // indirect
 	github.com/test-network-function/graphsolver-exports v0.0.0-20220909223441-46e11b864674 // indirect
-	github.com/test-network-function/l2discovery-exports v0.0.0-20220929081721-4e9a42510716 // indirect
+	github.com/test-network-function/l2discovery-exports v0.0.0-20220926190927-5f72a82c96d4 // indirect
```
Comment (aneeshkp) [inline go.mod:78]: This should be upgraded to 
v0.0.0-20220926190927-5f72a82c96d4 => v0.0.0-20220929081721-4e9a42510716
or else it error's out 
 ptpInterface.IfPci.Description undefined

Comment (aneeshkp) [reply go.mod:78]: need to update go mod vendor

**Review by aneeshkp — APPROVED** [review]

**Pull request merged**